plasmapy/simulation/tests/test_particle_tracker.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tests for particle_tracker.py
+"""
+import astropy.units as u
+import numpy as np
+import pytest
+
+from hypothesis import given, settings
+from hypothesis import strategies as st
+from hypothesis.extra.numpy import arrays
+
+from plasmapy.formulary.lengths import gyroradius
+from plasmapy.particles import CustomParticle
+from plasmapy.plasma.grids import CartesianGrid
+from plasmapy.simulation.particle_tracker import (
+    MemoryIntervalSaveRoutine,
+    NoFieldsStoppingCondition,
+    ParticleTracker,
+    TimeElapsedStopCondition,
+)
+
+
+class TestParticleTrackerGyroradius:
+    point_particle = CustomParticle(1 * u.kg, 1 * u.C)
+
+    def instantiate_simulation(self, B_strength=1 * u.T):
+        L = 1e6 * u.km
+        num = 2
+        grid = CartesianGrid(-L, L, num=num)
+        grid_shape = (num,) * 3
+
+        Bz = np.full(grid_shape, B_strength) * u.T
+        grid.add_quantities(B_z=Bz)
+
+        return ParticleTracker(
+            grid, req_quantities=["E_x", "E_y", "E_z", "B_x", "B_y", "B_z"]
+        )
+
+    @pytest.mark.slow()
+    @given(arrays(int, (5,), elements=st.integers(1, 100)), st.integers(1, 100))
+    @settings(deadline=5e4, max_examples=3)
+    def test_gyroradius(self, v_x, B_strength):
+        """Test to ensure particles maintain their gyroradius over time"""
+        v_x = v_x * u.m / u.s
+        v = np.array([[v_x_element.value, 0, 0] for v_x_element in v_x]) * u.m / u.s
+
+        # Set the initial position to the gyroradius
+        # This means the particle will orbit the origin
+        R_L = gyroradius(B_strength, self.point_particle, Vperp=v_x)
+        x = np.array([[0, R_L_element.value, 0] for R_L_element in R_L]) * u.m
+
+        simulation = self.instantiate_simulation(B_strength=B_strength)
+        simulation.load_particles(x, v, self.point_particle)
+
+        stop_condition = TimeElapsedStopCondition(10 * u.s)
+        save_routine = MemoryIntervalSaveRoutine(0.1 * u.s)
+
+        simulation.run(stop_condition, save_routine, dt=1e-3 * u.s)
+
+        positions = np.asarray(save_routine.x_all) * u.m
+        distances = np.linalg.norm(positions, axis=-1)
+
+        assert np.isclose(distances, R_L, rtol=1e-2).all()
+
+    @pytest.mark.slow()
+    @given(arrays(int, (5,), elements=st.integers(1, 100)))
+    @settings(deadline=1e5, max_examples=3)
+    def test_kinetic_energy(self, v_x):
```

Comment:
I would run the simulation once (maybe in the instantiate method) then just run this test on the same output as the previous test. No need to run it again.